src/Common/Core/Impl/OS/Win32Process.cs

```diff
@@ -117,43 +129,48 @@ public class Win32Process {
             SECURITY_ATTRIBUTES processAttr = CreateSecurityAttributes(winIdentity == null ? WellKnownSidType.AuthenticatedUserSid : WellKnownSidType.NetworkServiceSid);
             SECURITY_ATTRIBUTES threadAttr = CreateSecurityAttributes(winIdentity == null ? WellKnownSidType.AuthenticatedUserSid : WellKnownSidType.NetworkServiceSid);
 
-            PROCESS_INFORMATION pi;
-            try {
-                if(winIdentity == null) {
-                    if(!CreateProcess(applicationName, commandLine, ref processAttr, ref threadAttr, true,
-                        (uint)(CREATE_PROCESS_FLAGS.CREATE_UNICODE_ENVIRONMENT | CREATE_PROCESS_FLAGS.CREATE_NO_WINDOW),
-                        environment.NativeEnvironmentBlock,
-                        workingDirectory,
-                        ref si,
-                        out pi)) {
-                        throw new Win32Exception(Marshal.GetLastWin32Error());
+            lock (_createProcessLock) {
```

Comment:
I was using the lock to keep the Error mode in sync. Error mode can impact how the system handles critical errors for the entire process. In this particular case SetThreadErrorMode does not work. Hence i had to use a process wide setting.